Source/CarthageKit/Xcode.swift

```diff
@@ -52,18 +52,69 @@ private func parseSwiftVersionCommand(output: String?) -> String? {
 
 /// Determines the Swift version of a framework at a given `URL`.
 internal func frameworkSwiftVersion(_ frameworkURL: URL) -> SignalProducer<String, SwiftVersionError> {
+
 	guard
 		let swiftHeaderURL = frameworkURL.swiftHeaderURL(),
 		let data = try? Data(contentsOf: swiftHeaderURL),
 		let contents = String(data: data, encoding: .utf8),
 		let swiftVersion = parseSwiftVersionCommand(output: contents)
 		else {
-			return SignalProducer(error: .unknownFrameworkSwiftVersion)
+			return SignalProducer(error: .unknownFrameworkSwiftVersion(message: "Could not derive version from header file."))
 	}
 
 	return SignalProducer(value: swiftVersion)
 }
 
+internal func dSYMSwiftVersion(_ dSYMURL: URL) -> SignalProducer<String, SwiftVersionError> {
+
+	// Pick one architecture
+	guard let arch = architecturesInPackage(dSYMURL).first()?.value else {
+		return SignalProducer(error: .unknownFrameworkSwiftVersion(message: "No architectures found in dSYM."))
+	}
+
+	// Check the .debug_info section left from the compiler in the dSYM.
+	let task = Task("/usr/bin/xcrun", arguments: ["dwarfdump", "--arch=\(arch)", "--debug-info", dSYMURL.path])
```

Comment:
This can take quite some time to execute, so to speed things up I decided to just execute it for one architecture. Which one doesn't matter.